compiler/src/org.graalvm.compiler.core/src/org/graalvm/compiler/core/gen/NodeLIRBuilder.java

```diff
@@ -419,11 +415,23 @@ public void doBlock(Block block, StructuredGraph graph, BlockMap<List<Node>> blo
         }
     }
 
+    @Override
     @SuppressWarnings("try")
-    protected void matchComplexExpressions(List<Node> nodes) {
+    public void matchBlock(Block block, StructuredGraph graph, StructuredGraph.ScheduleResult schedule) {
+        try (BlockScope blockScope = gen.getBlockScopeMatch(block)) {
```

Comment:
I don't think using BlockScope for this adds anything.  Seems like BlockScope could be called MatchScope and directly subclass AutoCloseable.  getBlockScopeMatch becomes getMatchScope.  I don't think MatchScope even need to be a visible type since it can just return AutoCloseable.

Reply:
If MatchScope is not visible and matchBlock() uses AutoCloseable then it expects AutoCloseable.close() to throw Exception and something needs to be done with the exception eventhough it's never actually thrown. Exposing MatchBlock seems cleaner then.

Reply:
Sorry I meant to suggest DebugCloseable which is our AutoCloseable that doesn't throw.

Reply:
Ok. I'm making that change.